### PR TITLE
Add 1.11.0 CPU and GPU Dockerfile

### PR DESCRIPTION
**Pull request by mvsusp**
- added TensorFlow 1.11.0 Dockerfiles
- reduce CPU compressed image size before 826.40 now 384.74
- reduce GPU compressed image size before 2163.95 now 1495.48

- NCCL version: 2.3.5-2+cuda9.0
- CDNN version: 7.3.1.20-1+cuda9.0
- TENSORRT version: 4.1.2

## List of CPU layers

CREATED BY                                       |     SIZE
------------------------------------------------|-------------
/bin/sh -c #(nop)  ENTRYPOINT ["entry.py"]      |     0B
|2 framework_installable=tensorflow-1.11.0-c…    |     805MB
/bin/sh -c #(nop) COPY file:c50a662d35592945…    |     21.6kB
/bin/sh -c #(nop) COPY file:a13e618a0c436a0e…    |     102MB
/bin/sh -c #(nop) COPY file:9f52b10d55d79f79…    |     103MB
/bin/sh -c #(nop) WORKDIR /                      |     0B
|2 framework_installable=tensorflow-1.11.0-c…    |     5.91MB
/bin/sh -c #(nop)  ENV PYTHONDONTWRITEBYTECO…    |     0B
|2 framework_installable=tensorflow-1.11.0-c…    |     85.6MB
/bin/sh -c #(nop)  ARG framework_support_ins…    |     0B
/bin/sh -c #(nop)  ARG framework_installable     |     0B
/bin/sh -c #(nop)  LABEL maintainer=Amazon AI    |     0B
/bin/sh -c #(nop)  CMD ["/bin/bash"]             |     0B
/bin/sh -c mkdir -p /run/systemd && echo 'do…    |     7B
/bin/sh -c sed -i 's/^#\s*\(deb.*universe\)$…    |     2.76kB
/bin/sh -c rm -rf /var/lib/apt/lists/*           |     0B
/bin/sh -c set -xe   && echo '#!/bin/sh' > /…    |     745B
/bin/sh -c #(nop) ADD file:a83ab1826f43e88bc…    |     115MB

## List of GPU layers

CREATED BY                                       |     SIZE
-------------------------------------------------|-----------
/bin/sh -c #(nop)  ENTRYPOINT ["entry.py"]       |     0B
|2 framework_installable=tensorflow_gpu-1.11…    |     1.33GB
/bin/sh -c #(nop) COPY file:c50a662d35592945…    |     21.6kB
/bin/sh -c #(nop) COPY file:ee00ca447133d070…    |     259MB
/bin/sh -c #(nop) COPY file:671509eb13fc5f1a…    |     746MB
/bin/sh -c #(nop) WORKDIR /                      |     0B
|2 framework_installable=tensorflow_gpu-1.11…    |     5.91MB
/bin/sh -c #(nop)  ENV PYTHONDONTWRITEBYTECO…    |     0B
|2 framework_installable=tensorflow_gpu-1.11…    |     1.37GB
/bin/sh -c #(nop)  ENV TF_TENSORRT_VERSION=4…    |     0B
/bin/sh -c #(nop)  ENV CUDNN_VERSION=7.3.1.2…    |     0B
/bin/sh -c #(nop)  ENV NCCL_VERSION=2.3.5-2+…    |     0B
/bin/sh -c #(nop)  ARG framework_support_ins…    |     0B
/bin/sh -c #(nop)  ARG framework_installable     |     0B
/bin/sh -c #(nop)  LABEL maintainer=Amazon AI    |     0B
/bin/sh -c #(nop)  ENV NVIDIA_REQUIRE_CUDA=c…    |     0B
/bin/sh -c #(nop)  ENV NVIDIA_DRIVER_CAPABIL…    |     0B
/bin/sh -c #(nop)  ENV NVIDIA_VISIBLE_DEVICE…    |     0B
/bin/sh -c #(nop)  ENV LD_LIBRARY_PATH=/usr/…    |     0B
/bin/sh -c #(nop)  ENV PATH=/usr/local/nvidi…    |     0B
/bin/sh -c echo "/usr/local/nvidia/lib" >> /…    |     46B
/bin/sh -c #(nop)  LABEL com.nvidia.cuda.ver…    |     0B
/bin/sh -c #(nop)  LABEL com.nvidia.volumes.…    |     0B
/bin/sh -c apt-get update && apt-get install…    |     1.53MB
/bin/sh -c #(nop)  ENV CUDA_PKG_VERSION=9-0=…    |     0B
/bin/sh -c #(nop)  ENV CUDA_VERSION=9.0.176      |     0B
/bin/sh -c apt-get update && apt-get install…    |     16.1MB
/bin/sh -c #(nop)  LABEL maintainer=NVIDIA C…    |     0B
/bin/sh -c #(nop)  CMD ["/bin/bash"]             |     0B
/bin/sh -c mkdir -p /run/systemd && echo 'do…    |     7B
/bin/sh -c sed -i 's/^#\s*\(deb.*universe\)$…    |     2.76kB
/bin/sh -c rm -rf /var/lib/apt/lists/*           |     0B
/bin/sh -c set -xe   && echo '#!/bin/sh' > /…    |     745B
/bin/sh -c #(nop) ADD file:a83ab1826f43e88bc…    |     115MB

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/integ/test_versions.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 
 @pytest.fixture
-def required_versions(framework_version):
+def required_versions(framework_version, processor):
```
Comment (ChoiByungWook) [inline test/integ/test_versions.py]: What does adding the processor do here?

File: docker/1.11.0/final/py2/Dockerfile.cpu [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    ca-certificates \
+    python \
+    curl \
+    nginx \
+ && rm -rf /var/lib/apt/lists/*
+
+# Python won’t try to write .pyc or .pyo files on the import of source modules
+ENV PYTHONDONTWRITEBYTECODE=1 PYTHONUNBUFFERED=1
+
+RUN set -ex; \
```
Comment (ChoiByungWook) [inline docker/1.11.0/final/py2/Dockerfile.cpu]: Just my opinion, but this spacing feels really choppy to me.

File: test/unit/test_experiment_trainer.py [hunk 3]
```diff
@@ -123,7 +123,7 @@ def test_configure_s3_file_system(os_env, botocore, boto_client, trainer):
     boto_client('s3', region_name=region).get_bucket_location.assert_called_once_with(Bucket='my')
 
     calls = [
-        call('S3_USE_HTTPS', '1'),
```
Comment (ChoiByungWook) [inline test/unit/test_experiment_trainer.py]: Why do we remove this line?

File: docker/1.11.0/final/py2/Dockerfile.cpu [hunk 4]
```diff
@@ -0,0 +1,66 @@
+FROM tensorflow/serving:1.11.0 as tensorflow_serving_image
```
[the remark this answers is not in the document]
Comment (nadiaya) [reply docker/1.11.0/final/py2/Dockerfile.cpu]: i believe official google images https://hub.docker.com/r/tensorflow/serving/

File: docker/1.11.0/final/py2/Dockerfile.cpu [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+			\( -type d -a \( -name test -o -name tests \) \) \
+			-o \
+			\( -type f -a \( -name '*.pyc' -o -name '*.pyo' \) \) \
+		\) -exec rm -rf '{}' +; \
+	rm -f get-pip.py
+
+WORKDIR /
+
+# Install TF Serving pkg
+COPY --from=tensorflow_serving_image /usr/bin/tensorflow_model_server /usr/bin/tensorflow_model_server
```
Comment (nadiaya) [inline docker/1.11.0/final/py2/Dockerfile.cpu]: Why is it better than installing it from repo like we were before:
https://github.com/aws/sagemaker-tensorflow-container/blob/master/docker/1.9.0/final/py2/Dockerfile.cpu#L54-L57

File: docker/1.11.0/final/py2/Dockerfile.cpu [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+COPY $framework_installable .
+COPY $framework_support_installable .
+
+RUN pip install --no-cache-dir \
```
Comment (nadiaya) [inline docker/1.11.0/final/py2/Dockerfile.cpu]: What is the size of this layer?

File: docker/1.11.0/final/py2/Dockerfile.cpu [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    numpy \
+    scipy \
+    sklearn \
+    pandas \
+    Pillow \
+    h5py \
+    tensorflow-serving-api==1.11.0 \
+    \
+    $framework_installable \
```
Comment (nadiaya) [inline docker/1.11.0/final/py2/Dockerfile.cpu]: before we had issue with tensorflow binaries already being installed as a serving dependency (as far as i remember).
Should we add --upgrade to make sure our mkl optimized binaries are installed or move framework installation earlier?

File: docker/1.11.0/final/py2/Dockerfile.gpu [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+    numpy \
+    scipy \
+    sklearn \
+    pandas \
+    Pillow \
+    h5py \
+    tensorflow-serving-api==1.11.0 \
+    \
+    $framework_installable \
+    $framework_support_installable \
```
Comment (nadiaya) [inline docker/1.11.0/final/py2/Dockerfile.gpu]: same question here: are we confident TF hadn't been installed yet.

File: docker/1.11.0/final/py2/Dockerfile.gpu [hunk 9]
```diff
@@ -0,0 +1,93 @@
+FROM tensorflow/serving:1.11.0-gpu as tensorflow_serving_image
```
Comment (nadiaya) [inline docker/1.11.0/final/py2/Dockerfile.gpu]: what are the final layers sizes?

**Pull request merged**